internal/transports/p2p/server.go

```diff
@@ -22,16 +22,15 @@ type server struct {
 }
 
 func NewServer(
-	config *config.P2PConfig,
-	chainParams *chaincfg.Params,
+	p2pConfig *config.P2PConfig,
 	headersService service.Headers,
 	chainService service.Chains,
 	log *zerolog.Logger,
 ) *server {
 	serverLogger := log.With().Str("service", "p2p-experimental").Logger()
 	server := &server{
-		config:         config,
-		chainParams:    chainParams,
+		config:         p2pConfig,
+		chainParams:    config.GetNetParams(p2pConfig.ChainNetType),
```

Comment:
I removed this param from constructor params, but I still think it should stay in the struct itself, because otherwise every time we'd want to use chainParams, we'd have to use the function GetNetParams(), which would be unnecessarily entering the `switch` statement every time.